atomisticparsers/h5md/parser.py

```diff
@@ -0,0 +1,787 @@
+                            property_values_dict['value_magnitude'] = value_magnitude
+
+                    bins = observable.get('bins')
+                    if bins is not None:
+                        bins_unit = bins.units if hasattr(bins, 'units') else None
+                        bins_magnitude = bins.magnitude if hasattr(bins, 'units') else bins
+                        if flag_known_property:
+                            property_values_dict['bins'] = bins_magnitude * bins_unit if bins_unit else bins_magnitude
+                        else:
+                            property_values_dict['bins_unit'] = str(bins_unit) if bins_unit else None
+                            property_values_dict['bins_magnitude'] = bins_magnitude
+
+                    property_dict[property_value_key].append(property_values_dict)
+                workflow_results[property_key].append(property_dict)
+
+        workflow_results = {}
+        ensemble_average_observables = self.observable_info.get('ensemble_average')
+        ensemble_property_dict = {
+            'property_type_key': 'ensemble_properties',
+            'property_type_value_key': 'ensemble_property_values',
+            'properties_known': {'radial_distribution_functions': 'radial_distribution_function_values'},
+            'property_keys_list': EnsembleProperty.__dict__.keys(),
+            'property_value_keys_list': EnsemblePropertyValues.__dict__.keys()
+        }
+        get_workflow_results(ensemble_property_dict, ensemble_average_observables, workflow_results)
+        correlation_function_observables = self.observable_info.get('correlation_function')
+        correlation_function_dict = {
+            'property_type_key': 'correlation_functions',
+            'property_type_value_key': 'correlation_function_values',
+            'properties_known': {'mean_squared_displacements': 'mean_squared_displacement_values'},
+            'property_keys_list': CorrelationFunction.__dict__.keys(),
+            'property_value_keys_list': CorrelationFunctionValues.__dict__.keys()
+        }
+        get_workflow_results(correlation_function_dict, correlation_function_observables, workflow_results)
+        self.parse_md_workflow(dict(method=workflow_parameters, results=workflow_results))
+
+    def init_parser(self):
+
+        self._data_parser.mainfile = self.filepath
+
+    def parse(self, filepath, archive, logger):
+
+        self.filepath = os.path.abspath(filepath)
+        self.archive = archive
+        self.logger = logging.getLogger(__name__) if logger is None else logger
+        self._maindir = os.path.dirname(self.filepath)
+        self._h5md_files = os.listdir(self._maindir)
+        self._basename = os.path.basename(filepath).rsplit('.', 1)[0]
+
+        self.init_parser()
+
+        if self._data_parser.filehdf5 is None:
+            return
+
+        sec_run = self.archive.m_create(Run)
```

Comment:
would be nice to rid or m_create. We will be using inheritances now so these m_create will not anymore work for generic sections like EntryData, ArchiveSection. do
```python
sec_run = Run()
self.archive.run.append(sec_run
```
This is one more cumbersome, but this makes it future-proof

Reply:
ok, I will avoid m_create from now on, thanks. done.